test/specs/new/lheading_following_table.html

```diff
@@ -0,0 +1,19 @@
+<table>
+  <thead>
+    <tr>
+      <th>abc</th>
+      <th>def</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>bar</td>
+      <td>foo</td>
+    </tr>
+    <tr>
+      <td>baz</td>
+      <td>boo</td>
+    </tr>
+  </tbody>
+</table>
+<h1 id="title">title</h1>
```

Comment:
GitHub seems to render this one differently

![image](https://user-images.githubusercontent.com/229881/74113403-dc70f780-4b71-11ea-9a86-a15e5055f8e5.png)

Reply:
Hmmm.... The spec says ["The table is broken at the first empty line, or **beginning of another block-level structure**"](https://github.github.com/gfm/#example-201), so that would seem to break their rule. If it's an issue, it's a simple matter of just removing the associated lines that check for `lheading`.

Reply:
Good point, your test seems correct. I also double checked to make sure our GFM spec tests included 201 `[extension] Tables` and it is passing.

Reply:
from the [spec](https://github.github.com/gfm/#setext-heading):

> In general, a setext heading need not be preceded or followed by a blank line. However, it cannot interrupt a paragraph, so when a setext heading comes after a paragraph, a blank line is needed between them.

Maybe since it cannot interrupt a paragraph it also shouldn't interrupt a table? I'm leaning toward following the implementation and posing this as an issue to [GitHub](https://github.com/github/cmark-gfm) to see what they have to say.

We could leave the regex in and just comment it out with a note so we can change it easily in the future.